dlrover/python/unified/backend/common/base_worker.py

```diff
@@ -77,55 +80,57 @@ def start(self):
                     f"Failed to instantiate user class {user_func}"
                 ) from e
             export_rpc_instance(None, inst)
-            if not hasattr(inst, "run"):
-                logger.error(
-                    f"User class {user_func} does not have a 'run' method, "
-                    "assert FINISH. This may cause job state inconsistency."
-                )
-                self._update_stage_force(
-                    WorkerStage.FINISHED, WorkerStage.READY
-                )
-                return
-
-            user_func = inst.run
+            user_func = getattr(inst, "run", None)
+
         logger.info(f"Exported RPC methods: {list(RPC_REGISTRY.keys())}")
+        self._user_rpc_ready.set()
+
+        if user_func is not None:
+            Thread(
+                target=self._execute_user_function,
+                args=(user_func,),
+                daemon=True,
+                name="user_main_thread",
+            ).start()
+        else:
+            logger.warning(
+                f"User class {user_func} does not have a 'run' method. "
+                "Assert this worker as SERVICER, ready to receive RPC calls."
+            )
+            self._on_execution_end(ExecutionResult.SERVICER)
```

Comment:
Are there any practical application scenarios where user_func is not present? 

If so, the value of a certain type field should be defined based on the presence or absence of user_func, and then the execution logic of on_execution_end should be determined through this type field. 

Currently, using ExecutionResult = servicer in this way is not appropriate(is a type, not a result).